C/c4Database.cc

```diff
@@ -103,6 +107,15 @@ C4Database* c4db_openAgain(C4Database* db,
     return c4db_open({path.data(), path.size()}, c4db_getConfig(db), outError);
 }
 
+bool c4db_copy(C4String sourcePath, C4String destinationPath, const C4DatabaseConfig* config,
+               C4Error *error) noexcept {
+    FilePath from(slice(sourcePath).asString());
```

Comment:
I'm nervous about leaving these lines outside the `try` block. There's nothing in their current implementation that explicitly throws exceptions, but they're allowed to, and `std::string` might conceivably throw `bad_alloc`.